cmd/umserver/main_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"database/sql"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/connect"
+	"github.com/segmentio/kafka-go"
+)
+
+func TestGracefulShutdown_Success(t *testing.T) {
+	relatedComponents := setComponents()
+
+	err := gracefulShutdown(3*time.Second, new(sync.WaitGroup), new(http.Server), relatedComponents)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGracefulShutdown_Fail(t *testing.T) {
+	relatedComponents := setComponents()
+
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	err := gracefulShutdown(1*time.Nanosecond, wg, new(http.Server), relatedComponents)
+	if err == nil {
+		t.Error("want error got nil")
+	}
+}
+
+func Test_GracefulShutdown_Timeout(t *testing.T) {
+	relatedComponents := setComponents()
+
+	t.Run("with timeout", func(t *testing.T) {
+		err := gracefulShutdown(0*time.Second, new(sync.WaitGroup), new(http.Server), relatedComponents)
+		if err.Error() != "timeout" {
+			t.Fatal("want timeout got nil")
+		}
+	})
+}
+
+// setComponents is using for setting database, consul and kafka as related component
+func setComponents() *closingStructure {
```

Comment:
You can create `io.Closer` mock for testing instead of `setComponents()`

Reply:
I made corrections.